src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -2,70 +2,104 @@
- *   but must have a unique simple name to avoid command line name collisions.
- * - should contain @Arguments for any values they wish to collect. @Arguments may be
- *   optional or required. If required, the arguments are in effect "provisionally
- *   required" in that they are contingent on the specific plugin being specified on
- *   the command line; they will only be marked by the command line parser as missing
- *   if the they have not been specified on the command line, and the plugin class
- *   containing the plugin argument *has* been specified on the command line (as
- *   determined by the command line parser via a call to isDependentArgumentAllowed).
- *
- * NOTE: plugin class @Arguments that are marked "optional=false" should be not have a primitive
- * type, and should not have an initial value, as the command line parser will interpret these as
- * having been set even if they have not been specified on the command line. Conversely, @Arguments
- * that are optional=true should have an initial value, since they parser will not require them
- * to be set in the command line.
- *
- * The methods for each descriptor are called in the following order:
- *
- *  getPluginClass()/getPackageNames() - once when argument parsing begins (if the descriptor
- *  has been passed to the command line parser as a target descriptor)
- *
- *  getClassFilter() - once for each plugin subclass found
- *  getInstance() - once for each plugin subclass that isn't filtered out by getClassFilter
- *  validateDependentArgumentAllowed  - once for each plugin argument value that has been
- *  specified on the command line for a plugin that is controlled by this descriptor
- *
- *  validateArguments() - once when argument parsing is complete
- *  getAllInstances() - whenever the pluggable class consumer wants the resulting plugin instances
- *
- *  getAllowedValuesForDescriptorArgument is only called when the command line parser is constructing
- *  a help/usage message.
+ * <p>
+ * Base class for a descriptor that defines a set of command line plugin classes that can be dynamically
+ * discovered by the command line parser. Plugin instances contain argument fields that extend the
+ * set of command line arguments accepted by the parser.
+ * </p>
+ * <p>
+ * Each plugin descriptor passed to the command line parser is queried to determine the class and
+ * package names to be searched for plugin classes controlled by that descriptor. The descriptor
+ * object itself is  added to the parser's list of argument sources, and for each matching plugin class
+ * discovered in the packages list defined the descriptor, the parser delegates back to the descriptor
+ * to obtain an instance of that plugin. Each of these plugin objects are then also added to the parsers's
+ * list of argument sources.
+ * </p>
+ * <p>
+ * Descriptor instances should have at least one @Argument field used to accumulate the plugins that
```

Comment:
Here you should mention that the pluginDescirptor is responsible for maintaining a list of instances discovered through `createInstanceForPlugin` for later validation and use, it is mentioned in some of the method comments below.

Reply:
Done.